oak-upgrade/src/test/java/org/apache/jackrabbit/oak/upgrade/cli/SegmentTarToSegmentAzureWithSASUriTest.java

```diff
@@ -0,0 +1,140 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jackrabbit.oak.upgrade.cli;
+
+import java.io.IOException;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Date;
+import java.util.EnumSet;
+
+import com.google.common.base.Joiner;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.SharedAccessBlobPermissions;
+import com.microsoft.azure.storage.blob.SharedAccessBlobPolicy;
+
+import org.apache.jackrabbit.oak.segment.azure.AzuriteDockerRule;
+
+import org.apache.jackrabbit.oak.spi.state.NodeStore;
+import org.apache.jackrabbit.oak.upgrade.cli.container.NodeStoreContainer;
+import org.apache.jackrabbit.oak.upgrade.cli.container.SegmentAzureNodeStoreContainer;
+import org.apache.jackrabbit.oak.upgrade.cli.container.SegmentTarNodeStoreContainer;
+import org.apache.jackrabbit.oak.upgrade.cli.parser.OptionParserFactory;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.READ;
+import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.WRITE;
+import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.ADD;
+import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.CREATE;
+import static com.microsoft.azure.storage.blob.SharedAccessBlobPermissions.LIST;
+
+
+public class SegmentTarToSegmentAzureWithSASUriTest extends AbstractOak2OakTest {
+
+    private static final Logger log = LoggerFactory.getLogger(SegmentTarToSegmentAzureWithSASUriTest.class);
+
+    public static final String DEFAULT_CONTAINER_NAME = "oak-test";
+
+    public static final String ACCOUNT_KEY = "AccountKey";
+
+    private final NodeStoreContainer source;
+
+    private final NodeStoreContainer destination;
+
+    public static final String KEY_SHARED_ACCESS_SIGNATURE = "SharedAccessSignature";
+
+    private  String sasToken;
+
+    private static final EnumSet<SharedAccessBlobPermissions> READ_WRITE = EnumSet.of(READ, LIST, CREATE, WRITE, ADD);
+
+    @ClassRule
+    public static AzuriteDockerRule azurite = new AzuriteDockerRule();
+
+    public SegmentTarToSegmentAzureWithSASUriTest() throws IOException {
+        source = new SegmentTarNodeStoreContainer();
+        destination = new SegmentAzureNodeStoreContainer(azurite,"repository");
+    }
+
+    @Override
+    protected NodeStoreContainer getSourceContainer() {
+        return source;
+    }
+
+    @Override
+    protected NodeStoreContainer getDestinationContainer() {
+        return destination;
+    }
+
+    @Before
+    public void prepare() throws Exception {
```

Comment:
This method overrides its superclass method and is duplicating a lot of code from there. I suggest to call the superclass method to avoid duplication:

```java
    @Override
    public void prepare() throws Exception {
        CloudBlobContainer container = azurite.getContainer(DEFAULT_CONTAINER_NAME);
        this.sasToken = container.generateSharedAccessSignature(policy(READ_WRITE), null);
        
        super.prepare();
    }
```
Adding the `@Override` annotation is also beneficial to make it clear that this method is actually an override.